src/egraph.rs

```diff
@@ -516,12 +521,17 @@ impl<L: Language, N: Analysis<L>> EGraph<L, N> {
             while let Some((node, class_id)) = self.analysis_pending.pop() {
                 let class_id = self.find_mut(class_id);
                 let node_data = N::make(self, &node);
+
                 let class = self.classes.get_mut(&class_id).unwrap();
-                match self.analysis.merge(&mut class.data, node_data) {
-                    Some(Ordering::Equal) | Some(Ordering::Greater) => {}
+                let data = std::mem::take(&mut class.data);
```

Comment:
I think the `take` is necessary, but it would be good if it wasn't. Specifically, while I'd like to be able to do this using the `entry` API on `self.classes`, I think that leads to problems -- we need to borrow the graph for the call to `N::modify(..)` as well as extending the `analysis_pending`, which needs to happen within the scope that the entry has already borrowed those.

Reply:
I was able to solve this in my commit.